### PR TITLE
Refs #3809 - Using defaults for AndOr cop

### PR DESCRIPTION
**Pull request by daviddavis**
Follow up PR to https://github.com/theforeman/foreman/pull/3634

### CODE REVIEW

File: app/controllers/api/base_controller.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ class BaseController < ActionController::Base
     rescue_from Foreman::MaintenanceException, :with => :service_unavailable
 
     def get_resource
-      instance_variable_get :"@#{resource_name}" or raise 'no resource loaded'
+      instance_variable_get(:"@#{resource_name}") || (raise 'no resource loaded')
```
Comment (domcleal) [inline app/controllers/api/base_controller.rb]: adding the brackets to the method call, i.e. `|| raise('no resource loaded')` is probably more typical

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -50,7 +50,7 @@ def current_user
 
   # Authorize the user for the requested action
   def authorize
-    (render :json => { :error => "Authentication error" }, :status => :unauthorized and return) unless User.current.present?
+    (render(:json => { :error => "Authentication error" }, :status => :unauthorized) && (return)) unless User.current.present?
```
Comment (domcleal) [inline app/controllers/application_controller.rb]: Please expand this out into multiple lines, it'll be clearer.

File: app/models/host/base.rb [hunk 3]
```diff
@@ -401,11 +401,14 @@ def set_interface(attributes, name, iface)
 
       logger.debug "Saving #{name} NIC for host #{self.name}"
       result = iface.save
-      result or begin
+
+      if !result
```
Comment (domcleal) [inline app/models/host/base.rb]: `unless`

File: app/models/host/base.rb [hunk 4]
```diff
@@ -401,11 +401,14 @@ def set_interface(attributes, name, iface)
...
         logger.warn "Saving #{name} NIC for host #{self.name} failed, skipping because:"
         iface.errors.full_messages.each { |e| logger.warn " #{e}"}
-        false
+        return false
```
Comment (domcleal) [inline app/models/host/base.rb]: superfluous? The `result` line below will return `false` anyway.

File: app/models/lookup_value.rb [hunk 5]
```diff
@@ -83,15 +83,15 @@ def ensure_fqdn_exists
     return md if md == true || md == false
     fqdn = md[1].split(LookupKey::KEY_DELM)[0]
     return true if Host.unscoped.find_by_name(fqdn) || host_or_hostgroup.try(:new_record?)
-    errors.add(:match, _("%{match} does not match an existing host") % { :match => "fqdn=#{fqdn}" }) and return false
+    errors.add(:match, _("%{match} does not match an existing host") % { :match => "fqdn=#{fqdn}" }) && (return false)
```
Comment (domcleal) [inline app/models/lookup_value.rb]: move the return onto a separate line

File: app/models/lookup_value.rb [hunk 6]
```diff
@@ -83,15 +83,15 @@ def ensure_fqdn_exists
...
   end
 
   def ensure_hostgroup_exists
     md = ensure_matcher(/hostgroup=(.*)/)
     return md if md == true || md == false
     hostgroup = md[1].split(LookupKey::KEY_DELM)[0]
     return true if Hostgroup.unscoped.find_by_name(hostgroup) || Hostgroup.unscoped.find_by_title(hostgroup) || host_or_hostgroup.try(:new_record?)
-    errors.add(:match, _("%{match} does not match an existing host group") % { :match => "hostgroup=#{hostgroup}" }) and return false
+    errors.add(:match, _("%{match} does not match an existing host group") % { :match => "hostgroup=#{hostgroup}" }) && (return false)
```
Comment (domcleal) [inline app/models/lookup_value.rb]: ditto

File: app/models/user.rb [hunk 7]
```diff
@@ -173,11 +173,11 @@ def select_title
   end
 
   def self.anonymous_admin
-    unscoped.find_by_login ANONYMOUS_ADMIN or raise Foreman::Exception.new(N_("Anonymous admin user %s is missing, run foreman-rake db:seed"), ANONYMOUS_ADMIN)
+    unscoped.find_by_login(ANONYMOUS_ADMIN) || (raise Foreman::Exception.new(N_("Anonymous admin user %s is missing, run foreman-rake db:seed"), ANONYMOUS_ADMIN))
```
Comment (domcleal) [inline app/models/user.rb]: ditto, move the brackets to `|| raise(..`

File: db/migrate/20110301154453_add_managed_to_hosts.rb [hunk 8]
```diff
@@ -5,7 +5,7 @@ def up
     Host.reset_column_information
 
     for host in Host.unscoped.all
-      host.update_attribute :managed, !!host.operatingsystem_id and !!host.architecture_id and (!!host.ptable_id or !host.disk.empty?)
+      host.update_attribute(:managed, !!host.operatingsystem_id) && !!host.architecture_id && (!!host.ptable_id || !host.disk.empty?)
```
Comment (domcleal) [inline db/migrate/20110301154453_add_managed_to_hosts.rb]: I think the brackets are in the correct place due to the operator precedence... but I don't think it's correct logically. I think the intention was to pass the whole expression as the second argument to `update_attribute`.

File: test/functional/compute_resources_vms_controller_test.rb [hunk 9]
```diff
@@ -177,7 +177,7 @@ def test_should_create_vm(name = "new_test")
   end
 
   def get_test_vm
-    @compute_resource.vms.index {|vm| vm.name == "test" and @test_vm = vm}
+    @compute_resource.vms.index {|vm| vm.name == "test" && (@test_vm = vm)}
```
Comment (domcleal) [inline test/functional/compute_resources_vms_controller_test.rb]: `@test_vm = @compute_resource.vms.find { |vm| vm.name == 'test' }`

**Comment by domcleal** [comment]
Please also rebase the PR.

**Comment by daviddavis** [comment]
[test]

**Comment by daviddavis** [comment]
@domcleal updated.

**Comment by domcleal** [comment]
Merged as e263719, thanks @daviddavis.

**Pull request closed without merging**